packages/react-core/src/scalprum-provider.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+import { initialize, AppsConfig } from '@scalprum/core';
+import { ScalprumContext } from './scalprum-context';
+
+/**
+ * This is totally random implemenetaion of something that does not exists yet. I needed some react code to setup tests.
+ */
+
+export type ScalprumFeed = AppsConfig | (() => AppsConfig) | (() => Promise<AppsConfig>);
+
+export interface ScalprumState<T = Record<string, unknown>> {
+  initialized: boolean;
+  config: AppsConfig;
+  api?: T;
+}
+
+export interface ScalprumProviderProps<T = Record<string, unknown>> {
+  config: ScalprumFeed;
+  api?: T;
+  children: React.ReactNode;
+}
+
+export function ScalprumProvider<T = Record<string, unknown>>({
+  config,
+  children,
+  api,
+}: ScalprumProviderProps): React.ReactElement | React.ReactElement {
+  const [state, setState] = useState<ScalprumState<T>>({
+    initialized: false,
+    config: {},
+    api: api as T,
+  });
+  useEffect(() => {
+    if (typeof config === 'object') {
+      initialize<T>({ scalpLets: config as AppsConfig, api: api as T });
+      setState((prev) => ({ ...prev, initialized: true, config: config as AppsConfig }));
+    }
+
+    if (typeof config === 'function') {
+      Promise.resolve(config()).then((config) => {
+        setState((prev) => ({ ...prev, initialized: true, config: config as AppsConfig }));
+        initialize<T>({ scalpLets: config as AppsConfig, api: api as T });
+      });
+    }
```

Comment:
You can merge these 2 together, if we'll resolve object it will work as if it's a function. I would also prefer to use `async` and `await` in here
```TS
(async () => {
  const result = (await Promise.resolve(config?.() || config)) as AppsConfig;
  setState((prev) => ({ ...prev, initialized: true, config: result  }));
  initialize<T>({ scalpLets: result, api: api as T });
})()
```

Reply:
Not merging it. Using async/await directly in effects is not a good idea. It postpones the whole update cycle. That's why there is the promise.

Reply:
That's why it's wrapped in IIFE. It has the same effect as using `then`, but there is no callback.